From cf602bcd3348fe493288bec221641783a465aa91 Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 13 May 2016 14:31:20 +0000
Subject: [PATCH 0745/1076] MANTIS 41103 Mantis 41103: CfgValues should mecker
 CdfgValues 1.1.4 Set an exit code, and display "could not read", if a file
 could not be read.

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1012 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/CfgTools/inc/CfgValues_def.h   | 12 ++-
 .../Massai/cpp/CfgTools/src/CfgValues.cpp     | 88 +++++++++++++------
 2 files changed, 70 insertions(+), 30 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/CfgTools/inc/CfgValues_def.h b/SelfServiceCommon/Massai/cpp/CfgTools/inc/CfgValues_def.h
index 1e3b99b8..1b4447f0 100644
--- a/SelfServiceCommon/Massai/cpp/CfgTools/inc/CfgValues_def.h
+++ b/SelfServiceCommon/Massai/cpp/CfgTools/inc/CfgValues_def.h
@@ -66,9 +66,15 @@
 // #define FILE_VER_DTS "1.1.2.0\0"
 //
 // 10.01.2014  GiT       - New functions: erase, removefromlist
-//
-#define FILE_VER_BIN 1,1,3,0
-#define FILE_VER_DTS "1.1.3.0\0"
+// #define FILE_VER_BIN 1,1,3,0
+// #define FILE_VER_DTS "1.1.3.0\0"
+//
+// 09.Dec.2009  JK      - Mantis 41103: CfgValues should mecker
+//                        Set an exit code, and display "could not read",
+//                        if a file could not be read.
+#define FILE_VER_BIN 1,1,4,0
+#define FILE_VER_DTS "1.1.4.0\0"
+
 //-------------------------------------------------------------------------
 
 
diff --git a/SelfServiceCommon/Massai/cpp/CfgTools/src/CfgValues.cpp b/SelfServiceCommon/Massai/cpp/CfgTools/src/CfgValues.cpp
index 41f3a2ae..e2793316 100644
--- a/SelfServiceCommon/Massai/cpp/CfgTools/src/CfgValues.cpp
+++ b/SelfServiceCommon/Massai/cpp/CfgTools/src/CfgValues.cpp
@@ -7,7 +7,7 @@
  *
  * @author Juergen Kreierhoff
  *
- * Copyright (c) 2004-2009 MATERNA Information & Communications
+ * Copyright (c) 2004-2016 MATERNA Information & Communications
  *
  **************************************************************************
 **/
@@ -43,6 +43,32 @@
 
 
 
+namespace
+{
+   
+   
+   int const c_exit_good = 0;
+   
+   
+   int const c_exit_some_items_failed = 1;
+   
+   
+   int const c_exit_rejected = 2;
+
+   
+   int const c_exit_connection_failed = 3;
+   
+   
+   int const c_exit_wrong_parameter = 4;
+   
+   
+   int const c_exit_file_not_read = 5;
+   
+   
+} // namespace
+
+
+
 /**
  *
  * Generate ATSAVER string.
@@ -699,11 +725,11 @@ int main( int argc, char ** argv )
             std::cout
                << DSPLERR "Please specfiy a filename!"
                << std::endl;
-            return 1;
+            return c_exit_wrong_parameter;
          }
          std::cout
-            << DSPLLIN "Reading configuration value(s) to file ["
-            << DstFile << "] ..." << std::endl;
+            << DSPLLIN "Reading configuration value(s) to file \""
+            << DstFile << "\" ..." << std::endl;
          //
          std::ofstream OS( DstFile );
          //
@@ -726,7 +752,7 @@ int main( int argc, char ** argv )
                << DSPLERR "Could not connect the "
                << CFGMGR_SRVNAME_DIRECTORY << " servant."
                << std::endl;
-            return 1;
+            return c_exit_connection_failed;
          }
          Config::access Acc;
          Acc.Item = CORBA::string_dup( ArgPath );
@@ -745,7 +771,7 @@ int main( int argc, char ** argv )
                      << DSPLERR "Could not connect the "
                      << CFGMGR_SRVNAME_DIRECTORY << " servant."
                      << std::endl;
-                  return 1;
+                  return c_exit_connection_failed;
                }
                //
                unsigned Records = 0;
@@ -767,7 +793,7 @@ int main( int argc, char ** argv )
                      << DSPLERR "Could not connect the "
                      << CFGMGR_SRVNAME_TEXT << " servant."
                      << std::endl;
-                  return 1;
+                  return c_exit_connection_failed;
                }
                anyItemFirst( OS, ArgPath, CFGTOKEN_TYPE_TEXT );
                readTextCopy( OS, TextProxy->get( Acc ) );
@@ -788,7 +814,7 @@ int main( int argc, char ** argv )
                      << DSPLERR "Could not connect the "
                      << CFGMGR_SRVNAME_TEXTLIST << " servant."
                      << std::endl;
-                  return 1;
+                  return c_exit_connection_failed;
                }
                if( TextlistProxy->get( Acc ) )
                {
@@ -819,7 +845,7 @@ int main( int argc, char ** argv )
                      << DSPLERR "Could not connect the "
                      << CFGMGR_SRVNAME_NUMBER << " servant."
                      << std::endl;
-                  return 1;
+                  return c_exit_connection_failed;
                }
                anyItemFirst( OS, ArgPath, CFGTOKEN_TYPE_NUMBER );
                OS << NumberProxy->get( Acc );
@@ -840,7 +866,7 @@ int main( int argc, char ** argv )
                      << DSPLERR "Could not connect the "
                      << CFGMGR_SRVNAME_SELECTION << " servant."
                      << std::endl;
-                  return 1;
+                  return c_exit_connection_failed;
                }
                anyItemFirst( OS, ArgPath, CFGTOKEN_TYPE_SELECTION );
                OS << SelectionProxy->getString( Acc );
@@ -850,7 +876,7 @@ int main( int argc, char ** argv )
 
          }
          //
-         return 0;
+         return c_exit_good;
       }
       //
       // Then try, if the WRITE-option is given
@@ -884,10 +910,18 @@ int main( int argc, char ** argv )
          Config::description CDescr;
          SAcc.Authorization = CORBA::string_dup( ACC_NAME );
 
-         std::cout
-            << DSPLLIN "Writing configuration value(s) from file "
-            << SrcFile << " ..." << std::endl;
          ifstream IS( SrcFile );
+         if( !IS.good() )
+         {
+            // File not opened
+            std::cout
+               << DSPLLIN "Could not read file \""
+               << SrcFile << "\"" << std::endl;
+            return c_exit_file_not_read;
+         }
+         std::cout
+            << DSPLLIN "Writing configuration value(s) from file \""
+            << SrcFile << "\" ..." << std::endl;
          while( IS.good() )
          {
             cfgValuesFRead( FItm, IS, CFGTOKEN_ITEM_NAME );
@@ -926,7 +960,7 @@ int main( int argc, char ** argv )
                         << DSPLERR "Could not connect the "
                         << CFGMGR_SRVNAME_TEXT << " servant."
                         << std::endl;
-                     return 1;
+                     return c_exit_connection_failed;
                   }
                   //
                   std::string ValString;
@@ -1000,7 +1034,7 @@ int main( int argc, char ** argv )
                         << DSPLERR "Could not connect the "
                         << CFGMGR_SRVNAME_TEXTLIST << " servant."
                         << std::endl;
-                     return 1;
+                     return c_exit_connection_failed;
                   }
                   //
                   char const * const AddToListUnique =
@@ -1168,7 +1202,7 @@ int main( int argc, char ** argv )
                         << DSPLERR "Could not connect the "
                         << CFGMGR_SRVNAME_NUMBER << " servant."
                         << std::endl;
-                     return 1;
+                     return c_exit_connection_failed;
                   }
                   //
                   if( SetInitial )
@@ -1243,7 +1277,7 @@ int main( int argc, char ** argv )
                         << DSPLERR "Could not connect the "
                         << CFGMGR_SRVNAME_SELECTION << " servant."
                         << std::endl;
-                     return 1;
+                     return c_exit_connection_failed;
                   }
                   //
                   if( SetInitial )
@@ -1363,14 +1397,14 @@ int main( int argc, char ** argv )
             std::cout << " processed successfully, ";
             soutItems( SetRecordsFailed );
             std::cout << " failed." << std::endl;
-            return 1;
+            return c_exit_some_items_failed;
          }
          else
          {
             std::cout << DSPLLIN;
             soutItems( SetRecordsTotal );
             std::cout << " processed sucessfully." << std::endl;
-            return 0;
+            return c_exit_good;
          }
       }
       //
@@ -1391,7 +1425,7 @@ int main( int argc, char ** argv )
           std::cout
              << DSPLERR "Could not connect the Directory servant."
              << std::endl;
-          return 1;
+          return c_exit_connection_failed;
         }
         Config::access SAcc;
         SAcc.Authorization = CORBA::string_dup( ACC_NAME );
@@ -1407,7 +1441,7 @@ int main( int argc, char ** argv )
         std::cout
            << DSPLLIN "Erased [" << DelItem << "]"
            << std::endl;
-        return 0;
+        return c_exit_good;
       }
       //
       // Then try, if the REMOVEFROMLIST-option is given
@@ -1430,7 +1464,7 @@ int main( int argc, char ** argv )
              << DSPLERR "Could not connect the "
              << CFGMGR_SRVNAME_TEXTLIST << " servant."
              << std::endl;
-          return 1;
+          return c_exit_connection_failed;
         }
         try
         {
@@ -1473,7 +1507,7 @@ int main( int argc, char ** argv )
                    << DSPLERR "Unable to write changed list to "
                    << RemItem << " servant."
                    << std::endl;
-                return 1;
+                return c_exit_connection_failed;
               }
             } else
             {
@@ -1485,7 +1519,7 @@ int main( int argc, char ** argv )
         {
           std::cout << DSPLERR "REMOVEFROMLIST item [" << RemItem << "] not found - ignored." << std::endl;
         }
-        return 0;         
+        return c_exit_good;         
       }
       //
       // No explicit command - invalid syntax!
@@ -1535,7 +1569,7 @@ int main( int argc, char ** argv )
                         " max=\"..\"\n"
             << std::endl;
 
-      return 0;
+      return c_exit_good;
    }
    catch( Config::Node::notFound & )
    {
@@ -1577,7 +1611,7 @@ int main( int argc, char ** argv )
    {
       std::cout << DSPLERR "Got unknown exception" << std::endl;
    }
-   return 1;
+   return c_exit_rejected;
 }
 
 
-- 
2.41.0.windows.1

